core/tests/test_network.py

```diff
@@ -0,0 +1,27 @@
+from testcontainers.core.container import DockerContainer
+from testcontainers.core.docker_client import DockerClient
+from testcontainers.core.network import Network
+
+
+def test_network_gets_created_and_cleaned_up():
+    with Network() as network:
+        docker = DockerClient()
+        networks_list = docker.client.networks.list(network.name)
+        assert networks_list[0].name == network.name
+        assert networks_list[0].id == network.id
+    assert not docker.client.networks.list(network.name)
+
+
+def test_containers_can_communicate_over_network():
+    with Network() as network:
+        with DockerContainer("nginx:alpine-slim").with_name(
+                "alpine1").with_kwargs(network=network.name) as alpine1:
```

Comment:
Does it make sense to use `with_kwargs` here and not have an explicit API `with_network(Network)`?

Reply:
:+1: added in e4d9c19